modules/codegen/src/main/scala/com/twilio/guardrail/generators/EndpointsClientGenerator.scala

```diff
@@ -0,0 +1,462 @@
+package com.twilio.guardrail
+package generators
+
+import java.util.Locale
+
+import _root_.io.swagger.v3.oas.models.PathItem.HttpMethod
+import cats.arrow.FunctionK
+import cats.data.{ Ior, NonEmptyList }
+import cats.implicits._
+import com.twilio.guardrail.generators.syntax.Scala._
+import com.twilio.guardrail.protocol.terms.client._
+import com.twilio.guardrail.terms.RouteMeta
+import com.twilio.guardrail.languages.ScalaLanguage
+import com.twilio.guardrail.shims._
+import java.net.URI
+
+import scala.collection.JavaConverters._
+import scala.meta._
+
+object EndpointsClientGenerator {
+  object ClientTermInterp extends FunctionK[ClientTerm[ScalaLanguage, ?], Target] {
+
+    private[this] def toDashedCase(s: String): String = {
+      val lowercased =
+        "^([A-Z])".r.replaceAllIn(s, m => m.group(1).toLowerCase(Locale.US))
+      "([A-Z])".r
+        .replaceAllIn(lowercased, m => '-' +: m.group(1).toLowerCase(Locale.US))
+    }
```

Comment:
Tangential, but at some point it would be nice to put all the case converters in some common string-utils object.  Right now they're just all over the place.